sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -212,6 +219,29 @@ class SparkSession private(
     new SparkSession(sparkContext, Some(sharedState))
   }
 
+  /**
+   * :: Experimental ::
+   * Create an identical copy of this `SparkSession`, sharing the underlying `SparkContext`
+   * and cached data. All the state of this session (i.e. SQL configurations, temporary tables,
+   * registered functions) is also copied over.
+   * Changes to base session are not propagated to cloned session, cloned is independent
```

Comment:
The cloned session is independent of this session, that is, any changes in either session is not reflected in the other.